integration/docker/csi/alluxio/nodeserver.go

```diff
@@ -87,45 +120,192 @@ func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 	return &csi.NodePublishVolumeResponse{}, nil
 }
 
-func (ns *nodeServer) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpublishVolumeRequest) (*csi.NodeUnpublishVolumeResponse, error) {
+func bindMountGlobalMountPointToPodVolPath(req *csi.NodePublishVolumeRequest) (*csi.NodePublishVolumeResponse, error) {
 	targetPath := req.GetTargetPath()
+	stagingPath := req.GetStagingTargetPath()
+
+	notMnt, err := ensureMountPoint(targetPath)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	if !notMnt {
+		glog.V(4).Infoln("target path is already mounted")
+		return &csi.NodePublishVolumeResponse{}, nil
+	}
+
+	args := []string{"--bind", stagingPath, targetPath}
+	command := exec.Command("mount", args...)
+	glog.V(4).Infoln(command)
+	stdoutStderr, err := command.CombinedOutput()
+	glog.V(4).Infoln(string(stdoutStderr))
+	if err != nil {
+		if os.IsPermission(err) {
+			return nil, status.Error(codes.PermissionDenied, err.Error())
+		}
+		if strings.Contains(err.Error(), "invalid argument") {
+			return nil, status.Error(codes.InvalidArgument, err.Error())
+		}
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	return &csi.NodePublishVolumeResponse{}, nil
+}
 
-	command := exec.Command("/opt/alluxio/integration/fuse/bin/alluxio-fuse",
-		"unmount", targetPath,
-	)
+func (ns *nodeServer) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpublishVolumeRequest) (*csi.NodeUnpublishVolumeResponse, error) {
+	targetPath := req.GetTargetPath()
+	command := exec.Command("/opt/alluxio/integration/fuse/bin/alluxio-fuse", "umount", targetPath)
+	glog.V(4).Infoln(command)
 	stdoutStderr, err := command.CombinedOutput()
 	if err != nil {
 		glog.V(3).Infoln(err)
 	}
 	glog.V(4).Infoln(string(stdoutStderr))
 
-	err = mount.CleanupMountPoint(req.GetTargetPath(), mount.New(""), false)
+	err = mount.CleanupMountPoint(targetPath, mount.New(""), false)
 	if err != nil {
 		glog.V(3).Infoln(err)
+	} else {
+		glog.V(4).Infof("Succeed in unmounting %s", targetPath)
 	}
-
 	return &csi.NodeUnpublishVolumeResponse{}, nil
 }
 
-func (ns *nodeServer) NodeUnstageVolume(ctx context.Context, req *csi.NodeUnstageVolumeRequest) (*csi.NodeUnstageVolumeResponse, error) {
-	return &csi.NodeUnstageVolumeResponse{}, nil
+func (ns *nodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRequest) (*csi.NodeStageVolumeResponse, error) {
+	if req.GetVolumeContext()["mountInPod"] != "true" {
+		return &csi.NodeStageVolumeResponse{}, nil
+	}
+	ns.mutex.Lock()
+	defer ns.mutex.Unlock()
+
+	glog.V(4).Infoln("Creating Alluxio-fuse pod and mounting Alluxio to global mount point.")
+	fusePod, err := getAndCompleteFusePodObj(ns.nodeId, req)
+	if err != nil {
+		return nil, err
+	}
+	if _, err := ns.client.CoreV1().Pods(os.Getenv("NAMESPACE")).Create(fusePod); err != nil {
+		if strings.Contains(err.Error(), "already exists") {
```

Comment:
Using http code 409 conflict for this?

Reply:
http code exists in the http result object, and when `err` is not `nil`, we won't return the http result.

Reply:
How about use this method https://github.com/kubernetes/apimachinery/blob/080c0c77fab5f76acfa0c31e9e59411ecc861973/pkg/api/errors/errors.go#L540?